code/modules/antagonists/ert/ert.dm

```diff
@@ -34,12 +34,12 @@
 /datum/antagonist/ert/get_team()
 	return ert_team
 
-/datum/antagonist/ert/New()
-	. = ..()
-	name_source = GLOB.last_names
-
 /datum/antagonist/ert/proc/update_name()
-	owner.current.fully_replace_character_name(owner.current.real_name,"[role] [pick(name_source)]")
+	var/name
+	if (name = pick(name_source))
```

Comment:
You are using assignment here rather than comparison.
`name = pick(name_source)` assigns the returned value to name, and then if that value is anything but null this conditional block will return true.
This If also does nothing, so should be removed or modified in some way

Reply:
using the assigned value to check null and pick doesn't runtime with null list just returns null